File: dotnet/Dilithia.Sdk/Models/Credential.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// A verifiable credential issued on-chain.
/// </summary>
/// <param name="Id">The unique credential identifier.</param>
/// <param name="SchemaId">The schema this credential conforms to.</param>
/// <param name="Issuer">The address of the credential issuer.</param>
/// <param name="Holder">The address of the credential holder.</param>
/// <param name="Claims">The credential claims as key-value pairs.</param>
/// <param name="Revoked">Whether the credential has been revoked.</param>
public record Credential(string Id, string SchemaId, Address Issuer, Address Holder, Dictionary<string, string> Claims, bool Revoked);

File: java/src/main/java/org/dilithia/sdk/model/Credential.java
package org.dilithia.sdk.model;

/**
 * Represents an issued credential stored on-chain.
 *
 * @param commitment the credential commitment hash
 * @param issuer     the issuer address
 * @param holder     the holder address
 * @param schemaHash the schema hash this credential conforms to
 * @param status     the credential status (e.g. "active", "revoked")
 * @param revoked    whether the credential has been revoked
 */
public record Credential(
        String commitment,
        String issuer,
        String holder,
        String schemaHash,
        String status,
        boolean revoked
) {}
